atc/worker/placement.go

```diff
@@ -1,31 +1,101 @@
 package worker
 
 import (
+	"errors"
+	"fmt"
 	"math/rand"
+	"strings"
 	"time"
 
 	"code.cloudfoundry.org/lager"
 	"github.com/concourse/concourse/atc/db"
 )
 
+type ContainerPlacementStrategyOptions struct {
+	ContainerPlacementStrategy []string `long:"container-placement-strategy" default:"volume-locality" choice:"volume-locality" choice:"random" choice:"fewest-build-containers" choice:"limit-active-tasks" description:"Method by which a worker is selected during container placement. If multiple methods are specified, they will be applied in order. Random strategy is deprecated, it is still accepted, but takes no effective. Now, no strategy equals to random strategy."`
```

Comment:
I think we should actually get rid of the deprecation notice - although it's true that no strategy is the same as a random strategy, I think it's not possible to specify no strategy, given that there's a default of `volume-locality`.

So, I don't think we'll even want to deprecate `random`, since operators should be allowed to use completely random placement if they so choose - but maybe we can mention that chaining `random` is redundant

Reply:
OK, make sense.